Fix error message when creating views on temporary tables

RIC06X opened the pull request:
### Description

Previously when creating views on temporary tables, Babelfish would throw a general error message which does not explain the reason. This commit corrects the error message and keeps it the same as in T-SQL.


### Issues Resolved
Corrected the error message when creating views on temporary tables

### Test Scenarios Covered ###
* **Use case based -**
Create views on temporary tables

* **Negative test cases -**
Create views on regular tables

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

RIC06X committed:
Fix error message when creating views on temporary tables

Previously when creating views on temporary tables, Babelfish would
throw a general error message which does not explain the reason. This
commit corrects the error message and keeps it the same as in SQL
Server.

Task: BABEL-1645
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

xhfanhe commented on test/JDBC/input/BABEL-1645.sql:
```diff
@@ -0,0 +1,24 @@
+go
+
+DROP TABLE #t
+
```
Missing a GO here

xhfanhe commented on test/JDBC/expected/BABEL-1645.out:
```diff
@@ -0,0 +1,27 @@
+DROP TABLE test
+go
+
+
```
Missing the DROP for temp table

xhfanhe commented:
In the commit message, please add a line to specify the Jira number
```
<Commit title>

<Commit body>

Task: BABEL-1645
Signed-off-by: xxx
```

RIC06X committed:
Updated error class and Deleted select check functions

Since create view statement will always contains a select statement,
we don't need to check if we are in a select statement node. We also
change the error class to feature not supported in this commit.

Task: BABEL-1645
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

macwoody commented on contrib/babelfishpg_tsql/src/tsqlIface.cpp:
```diff
@@ -932,12 +934,24 @@ class tsqlSelectStatementMutator : public TSqlParserBaseListener
 public:
 	PLtsql_expr_query_mutator *mutator;
-
+	
```
you may not want these spaces

macwoody commented on contrib/babelfishpg_tsql/src/tsqlIface.cpp:
```diff
@@ -932,12 +934,24 @@ class tsqlSelectStatementMutator : public TSqlParserBaseListener
 
 	void exitSelect_statement(TSqlParser::Select_statementContext *ctx) override
-	{
+	{	
```
here as well

macwoody commented on contrib/babelfishpg_tsql/src/tsqlIface.cpp:
```diff
@@ -932,12 +934,24 @@ class tsqlSelectStatementMutator : public TSqlParserBaseListener
+	/* Corner case check. If a view is created on a temporary table, we should throw an exception.
+	 * Here we are setting up flags for later check.
+	 */ 
+	void enterCreate_or_alter_view(TSqlParser::Create_or_alter_viewContext *ctx)
```
you may want to implement them in tsqlBuilder instead

RIC06X committed:
Removed Extra Spaces

Task:BABEL-1645
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

RIC06X committed:
Changed static flag variable to non-static

We changed the flag variable, in_create_or_alter_view, from static
variable to a non static varible in tsqlSelectStatementMutator
clsass, and we also added another listener function,
exitTable_source_item(), to check the flag and table name. This
function will throw a exception when when creating views on temporary
tables

Task: BABEL-1645
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>
